pkg/manager/manager_test.go

```diff
@@ -214,197 +299,194 @@ func Test_LoadManagerBadNetworkStore(t *testing.T) {
 	updates[Test1Cont1ACont2ALeaf2B] = devicechangetypes.NewTypedValueFloat(ValueLeaf2B159)
-	deletes = append(deletes, Test1Cont1ACont2ALeaf2A)
-	deletes = append(deletes, Test1Cont1ACont2ALeaf2C)
+	updatesForDevice1, deletesForDevice1, deviceInfo := makeDeviceChanges(Device1, updates, deletes)
+
+	err := mgrTest.SetNewNetworkConfig(updatesForDevice1, deletesForDevice1, deviceInfo, "Testing")
+
+	// TODO - similar configs are currently not detected
+	t.Skip()
 
-	_, _, err = mgrTest.SetNetworkConfig("Device1", "", "", updates, deletes, "Testing")
 	assert.ErrorContains(t, err, "configurations found for")
 }
 
 func Test_SetSingleSimilarNetworkConfig(t *testing.T) {
 
-	mgrTest, _, _, _, _, _ := setUp(t)
+	mgrTest, _, _, _ := setUp(t)
 
 	updates := make(devicechangetypes.TypedValueMap)
-	deletes := make([]string, 0)
-
+	deletes := []string{Test1Cont1ACont2ALeaf2A, Test1Cont1ACont2ALeaf2C}
 	updates[Test1Cont1ACont2ALeaf2B] = devicechangetypes.NewTypedValueFloat(ValueLeaf2B159)
-	deletes = append(deletes, Test1Cont1ACont2ALeaf2A)
-	deletes = append(deletes, Test1Cont1ACont2ALeaf2C)
+	updatesForDevice, deletesForDevice, deviceInfo := makeDeviceChanges(Device1, updates, deletes)
 
-	changeID, configName, err := mgrTest.SetNetworkConfig("Device1", "", "", updates, deletes, "Testing")
+	err := mgrTest.SetNewNetworkConfig(updatesForDevice, deletesForDevice, deviceInfo, "Testing")
 	assert.NilError(t, err, "Similar config not found")
-	assert.Assert(t, changeID != nil)
-	assert.Assert(t, configName != nil)
 }
 
 func matchDeviceID(deviceID string, deviceName string) bool {
 	return strings.Contains(deviceID, deviceName)
 }
 
 func TestManager_GetAllDeviceIds(t *testing.T) {
-	mgrTest, _, configurationStoreTest, _, _, _ := setUp(t)
+	mgrTest, _, _, _ := setUp(t)
 
-	device2config, err := store.NewConfiguration("Device2", "1.0.0", "TestDevice",
-		[]change.ID{})
-	assert.NilError(t, err)
-	configurationStoreTest[device2config.Name] = *device2config
-	device3config, err := store.NewConfiguration("Device3", "1.0.0", "TestDevice",
-		[]change.ID{})
-	assert.NilError(t, err)
-	configurationStoreTest[device3config.Name] = *device3config
+	updates := make(devicechangetypes.TypedValueMap)
+	updates[Test1Cont1ACont2ALeaf2A] = devicechangetypes.NewTypedValueFloat(ValueLeaf2B314)
+	deletes := []string{Test1Cont1ACont2ALeaf2C}
+	updatesForDevice2, deletesForDevice2, deviceInfo2 := makeDeviceChanges("Device2", updates, deletes)
+	err := mgrTest.SetNewNetworkConfig(updatesForDevice2, deletesForDevice2, deviceInfo2, "Device2")
+	assert.NilError(t, err, "SetTargetConfig error")
+	updatesForDevice3, deletesForDevice3, deviceInfo3 := makeDeviceChanges("Device2", updates, deletes)
+	err = mgrTest.SetNewNetworkConfig(updatesForDevice3, deletesForDevice3, deviceInfo3, "Device3")
+	assert.NilError(t, err, "SetTargetConfig error")
 
 	deviceIds := mgrTest.GetAllDeviceIds()
+	// TODO - this doesn't see the new devices
```

Comment:
i know i'm being a PITA but can you explain this TODO better, so all of use can create issues for these and reference the line, uncomment the .skip() and verify the passing of new code ?

Reply:
Sure. The real issue is that the GetAllDeviceIds() func is not yet implemented for the Atomix stores. I'll modify the comment accordingly.